staging/src/k8s.io/apiserver/pkg/endpoints/apiserver_test.go

```diff
@@ -4368,6 +4368,22 @@ func runRequest(t *testing.T, path, verb string, data []byte, contentType string
 	return response
 }
 
+// runRequestOrDie is like runRequest, but used for benchmarking.
+func runRequestOrDie(path, verb string, data []byte, contentType string) *http.Response {
+	request, err := http.NewRequest(verb, path, bytes.NewBuffer(data))
+	if err != nil {
+		panic(err)
+	}
+	if contentType != "" {
+		request.Header.Set("Content-Type", contentType)
+	}
+	response, err := http.DefaultClient.Do(request)
+	if err != nil {
+		panic(err)
+	}
+	return response
+}
```

Comment:
I feel like we should return the error in the previous `runRequest` function, avoid the duplication, and the caller should decide whether they want to Fatal or panic.

But stepping back a notch, I think `panic` and `Fatal` have almost the same goal, so I'm guessing your problem is that you can't give `testing.B` to `runRequest`. You could merely change `runRequest` to take a `[TB](https://pkg.go.dev/testing#TB)`

Reply:
Ya, I was being dumb and forgot TB existed, changed it.